mlx_graphs/nn/conv/rel_conv.py

```diff
@@ -0,0 +1,327 @@
+
+        # augment messages with the boundary condition
+        message = mx.concatenate(
+            [message, boundary], axis=0
+        )  # (num_edges + num_nodes, batch_size, input_dim)
+
+        return message
+
+    def aggregate(self, messages, indices, edge_weight, dim_size) -> mx.array:
+        # augment aggregation index with self-loops for the boundary condition
+        index = mx.concatenate(
+            [indices, mx.arange(dim_size[0])]
+        )  # (num_edges + num_nodes,)
+        edge_weight = mx.concatenate([edge_weight, mx.ones(dim_size[0])])
+        shape = [1] * messages.ndim
+        shape[self.node_dim] = -1
+        edge_weight = edge_weight.reshape(shape)
+
+        if self.aggregate_func == "pna":
+            mean = scatter(
+                messages * edge_weight,
+                index,
+                axis=self.node_dim,
+                out_size=dim_size[0],
+                aggr="mean",
+            )
+            sq_mean = scatter(
+                messages**2 * edge_weight,
+                index,
+                axis=self.node_dim,
+                out_size=dim_size[0],
+                aggr="mean",
+            )
+            max = scatter(
+                messages * edge_weight,
+                index,
+                axis=self.node_dim,
+                out_size=dim_size[0],
+                aggr="max",
+            )
+            # scatter_min is not implemented in MLX-graphs
+            # min = scatter(
+            #     messages * edge_weight,
+            #     index, axis=self.node_dim,
+            #     out_size=dim_size[0],
+            #     aggr="min"
+            # )
+            std = mx.clip(sq_mean - mean**2, a_min=self.eps, a_max=None).sqrt()
+            features = mx.concatenate(
+                [mean[..., None], max[..., None], std[..., None]], axis=-1
+            )
+            features = features.flatten(-2)
+            if self.input_dims == 2:
+                features = features[:, None, :]
+            degree_out = degree(index, dim_size[0])[..., None, None]
+            scale = degree_out.log()
+            scale = scale / scale.mean()
+            scales = mx.concatenate(
+                [
+                    mx.ones_like(scale),
+                    scale,
+                    1 / mx.clip(scale, a_min=1e-2, a_max=None),
+                ],
+                axis=-1,
+            )
+            output = (features[..., None] * scales[:, :, None, :]).flatten(-2)
+
+            if self.input_dims == 2:
+                output = output.squeeze(1)
+        else:
+            output = scatter(
+                messages * edge_weight,
+                index,
+                axis=self.node_dim,
+                out_size=dim_size[0],
+                aggr=self.aggregate_func,
+            )
+
+        return output
+
+    def update_nodes(
+        self,
+        aggregated: mx.array,
+        old: mx.array,
+    ) -> mx.array:
```

Comment:
This seems to create some issues with `MessagePassing` as it may receive additional warts - if I run the test I get 

```
        # shapes: (|N| -> |N|)
>       output = self.update_nodes(
            aggregated=aggregated,
            **update_kwargs,
        )
E       TypeError: GeneralizedRelationalConv.update_nodes() got an unexpected keyword argument 'input'

mlx_graphs/nn/message_passing.py:97: TypeError
=========================================== short test summary info ===========================================
FAILED tests/nn/conv/test_rel_conv.py::test_rel_conv - TypeError: GeneralizedRelationalConv.update_nodes() got an unexpected keyword argument 'input'
```

Reply:
I get this error too.

Reply:
fixed https://github.com/mlx-graphs/mlx-graphs/pull/111/commits/63e55b84d9d736e5ff717955c214c0c7c672130e